tests/test_ipv4_api.py

```diff
@@ -0,0 +1,155 @@
+
+
+def test_public_ipv4_ipapi_co_success():
+    result = get_public_ipv4(api=IPv4API.IPAPI_CO, geo=True, timeout=40, max_retries=4, retry_delay=90)
+    assert result["status"]
+    assert is_ipv4(result["data"]["ip"])
+    assert set(result["data"].keys()) == DATA_ITEMS
+    assert result["data"]["api"] == "ipapi.co"
+
+
+def test_public_ipv4_ipleak_net_success():
+    result = get_public_ipv4(api=IPv4API.IPLEAK_NET, geo=True, timeout=40, max_retries=4, retry_delay=90)
+    assert result["status"]
+    assert is_ipv4(result["data"]["ip"])
+    assert set(result["data"].keys()) == DATA_ITEMS
+    assert result["data"]["api"] == "ipleak.net"
+
+
+def test_public_ipv4_my_ip_io_success():
+    result = get_public_ipv4(api=IPv4API.MY_IP_IO, geo=True, timeout=40, max_retries=4, retry_delay=90)
+    assert result["status"]
+    assert is_ipv4(result["data"]["ip"])
+    assert set(result["data"].keys()) == DATA_ITEMS
+    assert result["data"]["api"] == "my-ip.io"
+
+
+def test_public_ipv4_ifconfig_co_success():
+    result = get_public_ipv4(api=IPv4API.IFCONFIG_CO, geo=True, timeout=40, max_retries=4, retry_delay=90)
+    assert result["status"]
+    assert is_ipv4(result["data"]["ip"])
+    assert set(result["data"].keys()) == DATA_ITEMS
+    assert result["data"]["api"] == "ifconfig.co"
+
+
+
+def test_public_ipv4_myip_la_success():
+    result = get_public_ipv4(api=IPv4API.MYIP_LA, geo=True, timeout=40, max_retries=4, retry_delay=90)
+    assert result["status"]
+    assert is_ipv4(result["data"]["ip"])
+    assert set(result["data"].keys()) == DATA_ITEMS
+    assert result["data"]["api"] == "myip.la"
+
+
+def test_public_ipv4_ipquery_io_success():
+    result = get_public_ipv4(api=IPv4API.IPQUERY_IO, geo=True, timeout=40, max_retries=4, retry_delay=90)
+    assert result["status"]
+    assert is_ipv4(result["data"]["ip"])
+    assert set(result["data"].keys()) == DATA_ITEMS
+    assert result["data"]["api"] == "ipquery.io"
+
+
+def test_public_ipv4_ipwho_is_success():
+    result = get_public_ipv4(api=IPv4API.IPWHO_IS, geo=True, timeout=40, max_retries=4, retry_delay=90)
+    assert result["status"]
+    assert is_ipv4(result["data"]["ip"])
+    assert set(result["data"].keys()) == DATA_ITEMS
+    assert result["data"]["api"] == "ipwho.is"
+
+
+def test_public_ipv4_freeipapi_com_success():
+    result = get_public_ipv4(api=IPv4API.FREEIPAPI_COM, geo=True, timeout=40, max_retries=4, retry_delay=90)
+    assert result["status"]
+    assert is_ipv4(result["data"]["ip"])
+    assert set(result["data"].keys()) == DATA_ITEMS
+    assert result["data"]["api"] == "freeipapi.com"
+
+
+
+def test_public_ipv4_ip_api_com_success():
+    result = get_public_ipv4(api=IPv4API.IP_API_COM, geo=True, timeout=40, max_retries=4, retry_delay=90)
+    assert result["status"]
+    assert is_ipv4(result["data"]["ip"])
+    assert set(result["data"].keys()) == DATA_ITEMS
+    assert result["data"]["api"] == "ip-api.com"
+
+
+
+
```

Comment:
why there are more spaces between these api test versus others

Reply:
fixed in https://github.com/openscilab/ipspot/pull/69/commits/840c6b1037a37ed7a218c17298b6f1199fd9e69a